interfaces/catamorphism.ts

```diff
@@ -0,0 +1,3 @@
+export interface Catamorphism<A, B> {
```

Comment:
What if there are more than 2 values inside the user-implemented container?

Reply:
Any example?

Reply:
Response could have 3 states: Pending, Success, Error

Reply:
We could do it in this way: https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgYUxkwFtowALBAZxIB4A1TKBTFauUVJAEw4FckAayQQA7kgDaAXQB8cAN4AoOKrjoIAGx518sgBQA6Y8wDm1AFyK4kgNKIkcIcACeEdHCYs2MatKv6AG6YmvzAVl6s7HbSAJRwALzy+HAAvrFW+ADcSqlKSjzA2JrMeMWY1BwAspguAEbAdACC8ggkYJrAJMDsBESk5FS0dJJNADRwgRAIPHKKKmoa2roGAFb81DABwaHhcE3xSQQTokh7+ofJGQS5+YXlUGUllXAAoggwFGh0ADITAEqtdqdbq9QjEMhQSg0eiSP5wf5zZRqdRaHR6fSddBbOBBEJhKw-S7HOAsUwUHF43ZWf7E-DXfC3ApFEqPODlF7-YDUfiaGB0ADKEwAYkCOl0er4+hDBjCRlMZhMhXBhUiFqoluiDGAejxkKYAnSJjzsLhKtt8XsBUb1JgEJoLdSVXSGUzmQ8yhAkJs4CRag0rDV6o0kPwSA0oLIlH7g4ZNfp1XAAPQAKnWPoSKaTuJ2YWJubwAGo4ABGMaJ1OnPCZ7MXRLyAAMSlidxZpXZXp9wA+XygVnen2+XPQaB6uFeUCg0BOYYjUe7g6gcbRCZRqax8BruOJDfLa5TZIpiSzOct+ctcGLJebrY9He98EePL5NO5vP5ofDaABwBHjyQ46TtAUZPu+y7aKuaipjqvD6setY7nuUEpiaZocFuVJ5vWkwXleSGqKmWD2iiGHEgAtNesRAA

Reply:
Added